pkg/expr/evaluator.go

```diff
@@ -42,9 +42,6 @@ type (
 
 	// Validator validates a given expression
 	Validator interface {
```

Comment:
`TypeChecker` works for me. I think it makes sense to name the functions `Eval` and `Assert`:
- `type, err := checker.Eval(expr, attrs)`
- `err := checker.Assert(expr, attrs, TYPE_FOO)`